src/librustdoc/html/render.rs

```diff
@@ -253,16 +253,26 @@ struct IndexItem {
 /// A type used for the search index.
 struct Type {
     name: Option<String>,
+    // true both for Option<T> and T, false otherwise.
+    generic: bool,
+    ty_params: Box<Vec<Type>>,
```

Comment:
Because I didn't know any better :). It was `Box<Type>` initially to avoid infinite recursion, and when I changed to `Vec` I didn't reconsider the `Box`. Will change, thanks!